src/main/java/duke/StorageList.java

```diff
@@ -0,0 +1,152 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.task.Deadline;
+import duke.task.Task;
+
+/**
+ * StorageList is a class that stores a list of strings and
+ * provides methods to manipulate the list.
+ */
+public class StorageList {
+    private final List<Task> list = new ArrayList<>();
+
+    /**
+     * Adds a string to the list.
+     *
+     * @param t Task to be added to the list
+     */
+    public void add(Task t) {
+        list.add(t);
+    }
+
+    /**
+     * Returns the Task at the specified index.
+     *
+     * @param index Index of the Task to be returned
+     * @return Task at the specified index
+     * @throws DukeException Thrown when the index is out of bounds
+     */
+    public Task get(int index) throws DukeException {
+        try {
+            return list.get(index);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Invalid index.");
+        }
+    }
+
+    /**
+     * Marks the Task at the specified index as completed.
+     *
+     * @param i Index of the Task to be marked as completed
+     * @throws DukeException Thrown when the index is out of bounds
+     */
+    public void mark(int i) throws DukeException {
+        try {
+            list.get(i).mark();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Invalid index.");
+        }
+    }
+
+    /**
+     * Unmarks the Task at the specified index as completed.
+     *
+     * @param i Index of the Task to be unmarked as completed
+     * @throws DukeException Thrown when the index is out of bounds
+     */
+    public void unmark(int i) throws DukeException {
+        try {
+            list.get(i).unmark();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Invalid index.");
+        }
+    }
+
+    /**
+     * Returns the size of the list.
+     *
+     * @return Size of the list
+     */
+    public int getSize() {
+        return list.size();
+    }
+
+    /**
+     * Deletes the Task at the specified index.
+     *
+     * @param i Index of the Task to be deleted
+     * @throws DukeException Thrown when the index is out of bounds
+     */
+    public void delete(int i) throws DukeException {
+        try {
+            list.remove(i);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Invalid index.");
+        }
+    }
+
+    /**
+     * Resets the list of tasks.
+     */
+    public void reset() {
+        list.clear();
+    }
+
+    /**
```

Comment:
I like that you implemented several ways to filter the tasks!

Reply:
Thanks!